src/Playwright/Core/FormData.cs

```diff
@@ -51,13 +51,17 @@ private IFormData SetImpl(string name, object value)
 
     public IFormData Set(string name, FilePayload value) => SetImpl(name, value);
 
-    internal IList<object> ToProtocol()
+    internal IList<object> ToProtocol(bool throwWhenSerializingFilePayloads = false)
     {
         var output = new List<object>();
         foreach (var kvp in Values)
         {
             if (kvp.Value is FilePayload file)
             {
+                if (throwWhenSerializingFilePayloads)
+                {
+                    throw new PlaywrightException("Form requests don't support file payloads, use Multipart=formData instead.");
```

Comment:
Doesn't it serialize just file name in case of url encoded params? In java we just send it as a json object which should make it clear on the receiving end that something was wrong with the encoding.

Reply:
Java serializes to JSON:

https://github.com/microsoft/playwright-java/blob/4204096c2495addb4acd373531a234dc8c62f6ba/playwright/src/main/java/com/microsoft/playwright/impl/Serialization.java#L363

in JS we are more strict:
```
  form?: { [key: string]: string|number|boolean; };
  multipart?: { [key: string]: string|number|boolean|fs.ReadStream|FilePayload; };
so we simply not allow it.
```

see here:

https://github.com/microsoft/playwright/blob/adc9b2d39025af39c741bd0deef9f884d6473dae/packages/playwright-core/src/client/fetch.ts#L348-L349

In Python we probably throw protocol wise: https://github.com/microsoft/playwright-python/blob/42c0bf19d7ae415552172d7c04cdb7afd9dad7fb/playwright/_impl/_fetch.py#L343